ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-materialized-aggregates.json

```diff
@@ -2175,6 +2184,446 @@
           {"row":{"columns":["8", 14000, 15000, 1]}}
         ]}
       ]
+    },
+    {
+      "name": "fail on unsupported query feature: where key range",
+      "statements": [
+        "CREATE STREAM INPUT (ID INTEGER KEY, IGNORED INT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE AGGREGATE AS SELECT ID, COUNT(1) AS COUNT FROM INPUT GROUP BY ID;",
+        "SELECT * FROM AGGREGATE WHERE 0 < ID AND ID < 100;"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  false
+      },
+      "expectedError": {
+        "type": "io.confluent.ksql.rest.entity.KsqlStatementErrorMessage",
+        "message": "Bound on key columns '[`ID` INTEGER KEY]' must currently be '='.",
+        "status": 400
+      }
+    },
+    {
+      "name": "Generalized WHERE: key range",
```

Comment:
I think it might make sense to start generating an `EXPLAIN` for pull queries so that we know the physical plan that we decided to take (something similar to what we do for historical plans for persistent queries). That way we can make sure that the code changes we make don't accidentally result in regressions that change the execution in a negative way (e.g. if we had implemented primary key scans and this PR turned it into a table scan).

Obviously a bigger change so requires a follow-up, I'll create a ticket from this comment for context and we can plan that work as part of the normal planning process.

Reply:
Hehe I just read your response to Guozhang's comments. I think `EXPLAIN` is useful for both testing and usability.

Reply:
#6958